### PR TITLE
gh-118359: Add notes for Bdb.user_call

### PR DESCRIPTION
**Pull request by gaogaotiantian**
`argument_list` of `bdb.Bdb.user_call` will always be `None` now.

* Issue: gh-118359

----
📚 Documentation preview 📚: https://cpython-previews--118368.org.readthedocs.build/

### CODE REVIEW

**Comment by gaogaotiantian** [comment]
This is a rather simple one, @AlexWaygood could you maybe take a look at this? Thanks!

File: Doc/library/bdb.rst [hunk 1]
```diff
@@ -240,6 +240,11 @@ The :mod:`bdb` module also defines two classes:
       Called from :meth:`dispatch_call` if a break might stop inside the
       called function.
 
+      .. note::
```
Comment (serhiy-storchaka) [inline Doc/library/bdb.rst]: The `note` directive is too loud. If we don't want to attract special attention to an important note, we usually write a note as a plain text.

Should not there be a `versionchanged` directive for this change?

File: Doc/library/bdb.rst [hunk 2]
```diff
@@ -240,6 +240,11 @@ The :mod:`bdb` module also defines two classes:
...
+
+         *argument_list* is not used anymore and will always be :data:`None`.
```
Comment (serhiy-storchaka) [inline Doc/library/bdb.rst]: BTW, `Note` is usually formatted as ``` ``Note`` ```.

**Comment by AlexWaygood** [comment]
The "# XXX 'arg' is no longer used" comment referred to in https://github.com/python/cpython/issues/118359#issue-2267405937 appears to be this one:

https://github.com/python/cpython/blob/98739c9078d57f37df7d7b9d76ad32f7a92381a3/Lib/bdb.py#L126

The comment appears to have been there for 25 years. Just to make sure I understand correctly: does that mean that this parameter has been unused for 25 years?

File: Doc/library/bdb.rst (hunk 2 above)
Comment (gaogaotiantian) [reply Doc/library/bdb.rst]: You meant `None`?

**Comment by gaogaotiantian** [comment]
> The comment appears to have been there for 25 years. Just to make sure I understand correctly: does that mean that this parameter has been unused for 25 years?

Maybe. I know that argument is not used since I started playing with `sys.settrace`. `bdb` is a very old module that no one touches for a long time. Also there are very few libraries in the market that actually use it.

File: Doc/library/bdb.rst (hunk 1 above)
Comment (gaogaotiantian) [reply Doc/library/bdb.rst]: So we want to just add this as a normal text to the method. The `versionchanged` would probably be 2.x or 1.x, do we really want to search for the exact version?

Comment (AlexWaygood) [reply Doc/library/bdb.rst]: I agree with just adding as normal text, without the `.. note::` directive. I think if the parameter really has been unused for 25 years, it's okay to not have a `.. versionchanged` directive.

File: Doc/library/bdb.rst (hunk 2 above)
Comment (serhiy-storchaka) [reply Doc/library/bdb.rst]: Yes, `None`.

File: Doc/library/bdb.rst (hunk 1 above)
Comment (serhiy-storchaka) [reply Doc/library/bdb.rst]: We only add `versionadded` directives for 3.x changes. If it has been always unused in Python 3, no `versionadded` directive is needed. But mention this in the commit message.

**Review by serhiy-storchaka — APPROVED** [review]
LGTM.

File: Doc/library/bdb.rst (hunk 1 above)
Comment (gaogaotiantian) [reply Doc/library/bdb.rst]: The commit message into main is generated when the PR is squash merged so I don't think I can do that now.

Comment (AlexWaygood) [reply Doc/library/bdb.rst]: Either I or Serhiy (whoever merges this) will have the ability to edit the commit message when squash-merging the PR

**Review by AlexWaygood — APPROVED** [review]
Thanks!

**Pull request merged**